[docs] Create new page for Grafana Advisor (#106091)
grafana/grafana: docs/sources/administration/grafana-advisor/_index.md

---
title: Grafana Advisor
description: Learn more about Grafana Advisor, the app to monitor the health of your Grafana instance
weight: 300
labels:
  products:
    - oss
    - cloud
    - enterprise
  stage: experimental
keywords:
  - grafana
  - grafana advisor
  - monitoring
  - instance health
---

# Grafana Advisor

> {{< docs/experimental product="Grafana Advisor" featureFlag="grafanaAdvisor" >}}

Author: https://github.com/grafana/grafana/issues/106893

## Overview

Grafana Advisor is a monitoring tool that helps administrators keep their Grafana instances running smoothly and securely. It automatically performs regular health checks on your Grafana server, providing actionable insights and recommendations for maintaining optimal system performance.

{{< admonition type="note" >}}
Currently, Grafana Advisor performs regular checks on data sources, plugins, and your Grafana instance, but we're planning to expand its capabilities in future releases to cover more aspects of your Grafana environment.

You can suggest new checks and provide feedback through this [form](https://docs.google.com/forms/d/e/1FAIpQLSf8T-xMZauFXZ1uHw09OjZLT_AaiY-cl-hJGwC6Krkj0ThmZQ/viewform).
{{< /admonition >}}

> 

Author: adding it now :)

---
title: Grafana Advisor
description: Learn more about Grafana Advisor, the app to monitor the health of your Grafana instance
weight: 300
labels:
  products:
    - oss
    - cloud
    - enterprise
  stage: experimental
keywords:
  - grafana
  - grafana advisor
  - monitoring
  - instance health
---

# Grafana Advisor

{{< docs/experimental product="Grafana Advisor" featureFlag="grafanaAdvisor" >}}

## Overview

Grafana Advisor is a monitoring tool that helps administrators keep their Grafana instances running smoothly and securely. It automatically performs regular health checks on your Grafana server, providing actionable insights and recommendations for maintaining optimal system performance.

{{< admonition type="note" >}}

> Currently, Grafana Advisor only performs regular health checks on data sources, plugins, and your SSO settings, but we're planning to expand its capabilities in future releases to cover more aspects of your Grafana environment.

Reviewer: ```suggestion
Currently, Grafana Advisor performs regular checks on data sources, plugins, and your Grafana instance, but we're planning to expand its capabilities in future releases to cover more aspects of your Grafana environment.
```

You can suggest new checks and provide feedback through this [form](https://docs.google.com/forms/d/e/1FAIpQLSf8T-xMZauFXZ1uHw09OjZLT_AaiY-cl-hJGwC6Krkj0ThmZQ/viewform).
{{< /admonition >}}

## Before you begin

To set up Grafana Advisor you need:

- Administration rights in your Grafana organization.

> - If you're running Grafana on-premise, enable the required feature toggles in your Grafana instance. Refer to [Enable required feature toggles](#enable-feature-toggles) for instructions. This is not required if you're using Grafana Cloud, as the feature toggles are enabled by default.

Reviewer: ```suggestion
- If you're running Grafana on-premise, enable the required feature toggle in your Grafana instance. Refer to [Enable required feature toggles](#enable-feature-toggles) for instructions. This is not required if you're using Grafana Cloud, as the feature toggles are enabled by default.
```

---
title: Grafana Advisor
description: Learn more about Grafana Advisor, the app to monitor the health of your Grafana instance
weight: 300
labels:
  products:
    - oss
    - cloud
    - enterprise
  stage: experimental
keywords:
  - grafana
  - grafana advisor
  - monitoring
  - instance health
---

# Grafana Advisor

{{< docs/experimental product="Grafana Advisor" featureFlag="grafanaAdvisor" >}}

## Overview

> Grafana Advisor is a monitoring tool that helps administrators keep their Grafana instances running smoothly and securely. It automatically performs health checks on your Grafana server, providing actionable insights and recommendations for maintaining optimal system performance.

Reviewer 2: ```suggestion
Grafana Advisor is a monitoring tool that helps administrators keep their Grafana instances running smoothly and securely. It automatically performs regular health checks on your Grafana server, providing actionable insights and recommendations for maintaining optimal system performance.
```

{{< admonition type="note" >}}
Currently, Grafana Advisor only performs health checks on data sources, plugins, and your SSO settings, but we're planning to expand its capabilities in future releases to cover more aspects of your Grafana environment.

You can suggest new checks and provide feedback through this [form](https://docs.google.com/forms/d/e/1FAIpQLSf8T-xMZauFXZ1uHw09OjZLT_AaiY-cl-hJGwC6Krkj0ThmZQ/viewform).
{{< /admonition >}}

## Before you begin

To set up Grafana Advisor you need:

- Administration rights in your Grafana organization.
- Enable the required feature toggles in your Grafana instance. Refer to [Enable required feature toggles](#enable-feature-toggles) for instructions.

> ### Enable feature toggles

Reviewer 2: We should make it clear that this only applies to on-premise deployments.

To enable the required feature toggles, add them to your Grafana configuration file:

1. Open your Grafana configuration file, either `grafana.ini` or `custom.ini`. For file location based on the operating system, refer to Configuration file location.
1. Locate or add a `[feature_toggles]` section. Add this value:

   ```ini
   [feature_toggles]
   grafanaAdvisor = true
   ```

1. Save the changes to the file and restart Grafana.

## Access Grafana Advisor

1. Log in to your Grafana instance with an administrator account
1. Navigate to the Administration section
1. Select "Advisor" from the navigation menu

![<Grafana Advisor UI>](/media/docs/grafana-advisor/grafana-advisor-ui.png)

## Understand the Advisor interface

### Action needed section

This collapsible section displays issues requiring immediate attention:

- Each issue is listed with descriptive text outlining the problem
- For data source issues, Grafana Advisor displays the specific data source name.

> - A "Fix me" button appears. This button points you to the relevant issues to fix

Reviewer: The text and the buttons will change, depending on the issue type. I'd change this line with something like: 

```
 - For each item, Grafana Advisor displays the specific name of the item that needs to be fixed.
 - One or more buttons appear. These buttons point you to different links to fix the issue, retry the check or hide the error.
```

![Action needed](/media/docs/grafana-advisor/action_needed.png)

### Investigation needed section

> This collapsible section provides information on plugins that require an upgrade. Clicking an item opens the plugin's upgrade page. From there, you can either update to the latest version or select a specific version from the version history tab.

Reviewer 2: good catch

![Investigation needed](/media/docs/grafana-advisor/investigation-needed.png)

### More info section

This collapsible section provides details about the checks performed:

**Data source checks**

- Shows the number of data sources analyzed
- UID validation: Checks if the UID of each data source is valid

> - Health check: Verifies if data sources are responding properly

Reviewer 2: _Somewhere_ we need to indicate that these checks only apply to data sources with a backend... this could be here, in the Types of checks performed table, a combination of the two/somewhere else. Would welcome your input

Author: adding this info to the table!

- Missing plugin check: Verifies that each data source has its associated plugin installed

**Plugin Checks**

- Shows the number of plugins analyzed (for example, "X plugin(s) analyzed")
- Deprecation check: Identifies any installed plugins that are deprecated
- Update check: Detects if any installed plugins have newer versions available

**SSO Checks**

> - Checks if the list configurations in SSO settings are in a valid list format (space-separated, comma-separated, or JSON array).

Reviewer: This still needs to be removed, right?

| **Health check**         | Verify data source connectivity       | Checks if a data source is healthy and responding                 | <ul><li> Click "Fix me" to navigate to the data source instance <li> Click "Save and Test" to review the error message for the data source <li> Update the configuration to remediate the error. For example, it may be that the URL is incorrect or a token has expired and needs to be replaced</ul>                                                  |
| **Missing plugin check** | Confirm required plugins are present  | Checks if the plugin associated with the data source is installed | <ul><li>Review the list of data sources. Orphaned instances which are no longer required can be deleted via this list <li>If the corresponding plugin is available in the catalog, click "Install plugin" to navigate to the plugin details page and then click "Install"</ul>                                                                          |

> | **Deprecation check**    | Identify outdated components          | Checks if any installed plugins are deprecated                    | <ul><li>If a plugin has been deprecated, it means it's no longer going to receive updates and you should plan to migrate to an alternative <li>Navigate to the plugin details page to review the deprecation notice which may include guidance on alternatives <li>If none are specified, review the plugins catalog to discover potential options</ul> |

Reviewer 2: ```suggestion
| **Deprecation check**    | Identify outdated components          | Checks if any installed plugins are deprecated                    | <ul><li>If a plugin has been deprecated, it means it's no longer going to receive updates and you should plan to migrate to an alternative <li>Navigate to the plugin details page to review the deprecation notice which may include guidance on alternatives <li>If none are specified, review the [plugins catalog](https://grafana.com/grafana/plugins/) to discover potential options</ul> |
```

> | **Update check**         | Keep plugins current                  | Checks if an installed plugin has a newer version available       | <ul><li>We strongly recommend running the latest version of plugins <li> Click "Upgrade" to navigate to the plugin details page and either select "Update" to move to the latest version, or choose a specific version of the plugin within the version history tab</ul>                                                                                |

Reviewer 2: We do also have the SSO check to add here, but could always follow up with that

Author: @andresmgot could you help us out here? thank you in advance!
